pkg/kapis/devops/v1alpha2/devops.go

```diff
@@ -422,10 +424,12 @@ func (h *ProjectPipelineHandler) hasSubmitPermission(req *restful.Request) (hasP
 			}
 
 			for _, step := range node.Steps {
-				if step.ID != stepId || step.Input == nil {
+				if step.ID != stepId {
 					continue
 				}
-
+				if step.Input == nil {
+					err = errors.New("the step state was unapprovable")
```

Comment:
Suggestion:

`err = fmt.Errorf("the step '%s' is not approvable", step.DisplayName)`

Reply:
Done. Please see https://github.com/kubesphere/ks-devops/pull/374/commits/10c0284ca2946f5ddf224c7a7b6ae8f07579ed35.